File: app/src/main/java/com/hgianastasio/biblioulbrav2/models/user/RenewLoansResponseModelMapper.java
package com.hgianastasio.biblioulbrav2.models.user;

import com.hgianastasio.biblioulbrav2.core.base.boundaries.Mapper;
import com.hgianastasio.biblioulbrav2.core.user.models.RenewLoansResponse;

import javax.inject.Inject;

/**
 * Created by heitor_12 on 11/05/17.
 */

public class RenewLoansResponseModelMapper extends Mapper<RenewLoansResponse,RenewLoansResponseModel> {
    @Inject public RenewLoansResponseModelMapper() {}

    @Override
    public RenewLoansResponseModel transform(RenewLoansResponse input) {
        return new RenewLoansResponseModel(input.getStatus(),input.getMessage());
    }

    @Override
    public RenewLoansResponse transformBack(RenewLoansResponseModel input) {
        return new RenewLoansResponse(input.isSuccess(),input.getMessage());
    }
}

File: app/src/main/java/com/hgianastasio/biblioulbrav2/models/user/RenewLoansResponseModelMapper.kt
package com.hgianastasio.biblioulbrav2.models.user

import com.hgianastasio.biblioulbrav2.core.base.boundaries.Mapper
import com.hgianastasio.biblioulbrav2.core.user.models.RenewLoansResponse
import javax.inject.Inject

/**
 * Created by heitor_12 on 11/05/17.
 */
class RenewLoansResponseModelMapper @Inject constructor() : Mapper<RenewLoansResponse, RenewLoansResponseModel>() {
    override fun transform(input: RenewLoansResponse): RenewLoansResponseModel {
        return RenewLoansResponseModel(input.status, input.message.orEmpty())
    }

    override fun transformBack(input: RenewLoansResponseModel): RenewLoansResponse {
        return RenewLoansResponse(input.isSuccess, input.message)
    }
}